nemo_aligner/data/nlp/builders.py

```diff
@@ -265,7 +265,19 @@ def build_dataset(index, name):
 
 
 def build_sft_dataset(data_cfg, tokenizer, num_samples, answer_only_loss=True, is_chat=True, special_tokens=None):
-    dataset_cls = GPTSFTChatDataset if is_chat else GPTSFTDataset
+    packed_sequence = data_cfg.get("packed_sequence", False)
+    dataset_kwargs = {}
+
+    ## NOTE: sequence packing currently not supported with chat dataset
```

Comment:
I'd prefer us throwing a `NotImplementedError` in the case of packed_sequence=True and is_chat=True. Right now if someone does packed,is_chat=(True,True), they won't get an error, but the may be confused why they don't see any difference. 

We can also clean up that note if we raise the error, since it's self documenting